workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py

```diff
@@ -0,0 +1,104 @@
+import dataclasses
+from typing import (
+    Hashable,
+    Iterable,
+    MutableMapping,
+    Mapping,
+    Optional,
+    Sequence,
+    Set,
+    cast,
+)
+import fv3fit
+import xarray as xr
+import vcm
+import os
+
+
+State = MutableMapping[Hashable, xr.DataArray]
+
+
+@dataclasses.dataclass
+class MachineLearningConfig:
+    """Machine learning configurations
+
+    Attributes:
+        model: list of URLs to fv3fit models.
+        diagnostic_ml: do not apply ML tendencies if true.
+        scaling: if given, scale the outputs by the given factor. This is a manually
+            defined alteration of the model, and should not be used for
+            normalization.
+    Example::
+
+        MachineLearningConfig(
+            model=["gs://vcm-ml-scratch/test-annak/ml-pipeline-output"],
+            diagnostic_ml=False,
+        )
+
+    """
+
+    model: Sequence[str] = dataclasses.field(default_factory=list)
+    diagnostic_ml: bool = False
+    scaling: Mapping[str, float] = dataclasses.field(default_factory=dict)
+
+
+class MultiModelAdapter:
+    def __init__(
+        self,
+        models: Iterable[fv3fit.Predictor],
+        scaling: Optional[Mapping[str, float]] = None,
+    ):
+        """
+        Args:
+            models: models for which to combine predictions
+            scaling: if given, scale the predictions by the given factor
+        """
+        self.models = models
+        if scaling is None:
+            self._scaling: Mapping[str, float] = {}
+        else:
+            self._scaling = scaling
+
+    @property
+    def input_variables(self) -> Set[str]:
+        vars = [model.input_variables for model in self.models]
+        return {var for model_vars in vars for var in model_vars}  # type: ignore
+
+    def predict(self, arg: xr.Dataset) -> xr.Dataset:
+        predictions = []
+        for model in self.models:
+            predictions.append(model.predict(arg))
+        ds = xr.merge(predictions)
+        for var, scale in self._scaling.items():
+            ds[var] *= scale
+        return ds
+
+
+def open_model(config: MachineLearningConfig) -> MultiModelAdapter:
+    model_paths = config.model
+    models = []
+    for path in model_paths:
+        model = cast(fv3fit.Predictor, fv3fit.load(path))
+        models.append(model)
+    return MultiModelAdapter(models, scaling=config.scaling)
+
+
+def download_model(config: MachineLearningConfig, path: str) -> Sequence[str]:
```

Comment:
I know there are some temporary directory implementations to download and open models somewhere in fv3net rather than having to specify the path.  I don't know what context this function is used in, but consider if it's necessary to have the models in a specific place rather than removed upon completion of the run.

Reply:
Good point. This function is currently not used, and it's not clear yet how we are going to pass the pre-trained model to the actual scream run. For now, since this function is not tested, I will go ahead and remove. May add it back later if we end up needing to download the model first.